mlpipeline/encoder_service/app.py

```
@@ -0,0 +1,98 @@
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    HERE = Path(__file__).resolve().parent
+    ROOT = HERE.parent
+    ckpt_path = ROOT / "pretrain" / "checkpoints" / "user_encoder.pt"
+
+    if _MODEL is None or _INPUT_DIM != d:
+        if not ckpt_path.exists():
+            raise HTTPException(500, f"checkpoint not found: {ckpt_path}")
+        try:
+            _MODEL = _load_model(str(ckpt_path), input_dim=d, device=device)
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)